### PR TITLE
Pokemon optimizer enhancements

### PR DESCRIPTION
**Pull request by Anakin5**
- Add parameter "use_candies_for_xp" to activate/deactivate usage of candies to maximize xp.
- Add comments in the configuration example.
- Add dps, dps_attack and dps_defense in available sorting keys. So you can now keep the best move.
- Display ncp and dps for released and evolved Pokemons.
- Display Pokemon Bag count update

default config, should be good enough if you don't want to keep weak Pokemons
```json
{
    "type": "PokemonOptimizer",
    "config": {
        "transfer": true,
        "evolve": true,
        "use_candies_for_xp": true,
        "use_lucky_egg": true,
        "evolve_only_with_lucky_egg": true,
        "minimum_evolve_for_lucky_egg": 90,
        "keep": [
            {
                "top": 1,
                "evolve": true,
                "sort": ["iv"]
            },
            {
                "top": 1,
                "evolve": true,
                "sort": ["ncp"]
            },
            {
                "top": 1,
                "evolve": false,
                "sort": ["cp"]
            }
        ]
    }
},
```

or more aggressive, maybe once you have a lot of good pokemons and need stronger criteria to make a difference:

```json
{
    "type": "PokemonOptimizer",
    "config": {
        "transfer": true,
        "evolve": true,
        "use_candies_for_xp": true,
        "use_lucky_egg": true,
        "evolve_only_with_lucky_egg": true,
        "minimum_evolve_for_lucky_egg": 90,
        "keep": [
            {
                "top": 1,
                "evolve": true,
                "sort": ["iv", "dps_attack"]
            },
            {
                "top": 1,
                "evolve": true,
                "sort": ["iv", "dps_defense"]
            },
            {
                "top": 1,
                "evolve": true,
                "sort": ["ncp"]
            }
        ]
    }
},
```

### CODE REVIEW

File: pokemongo_bot/cell_workers/transfer_pokemon.py [hunk 1]
```diff
@@ -153,7 +153,9 @@ def release_pokemon(self, pokemon):
             data={
                 'pokemon': pokemon.name,
                 'cp': pokemon.cp,
-                'iv': pokemon.iv
+                'iv': pokemon.iv,
+                'ncp': '?',
```
Comment (TheSavior) [inline pokemongo_bot/cell_workers/transfer_pokemon.py]: Why do these need to be here but not need `xp` that you have in the other one?

**Comment by heihachi** [comment]
Why am i a potential reviewer? lol

**Comment by TheSavior** [comment]
:+1:

File: pokemongo_bot/cell_workers/transfer_pokemon.py (hunk 1 above)
Comment (Anakin5) [inline pokemongo_bot/cell_workers/transfer_pokemon.py]: You don't gain xp when exchanging pokemons. After we may have to choose what stats are the most relevant to display. I am displaying what I think people are interested to see to make sure they don't release good pokemons.

**Comment by nanogirl21** [comment]
@Anakin5 can we use '"evolve_speed": 5,` (or whatever speed number) with this code? Right now it's taking a very long time. Since this code waits after the entire bag is full to do release/evolve most people will have lots to do. Besides that everything works great. Thank you.

**Comment by nanogirl21** [comment]
@Anakin5 can we use '"evolve_speed": 1,' with this code? Right now it's taking a very long time. Since this code waits until the entire bag is 100% full to do release/evolve most people will have a lot. Today I released, used an egg for 1,000 XP, started to evolve, and half way through evolving my egg time (30 minutes) ran out. My XP it dropped back down to 500. I missed out on at least 15,000 XP when this happen. Besides tat that everything works great. Thank you.

**Comment by Anakin5** [comment]
I can add a parameter to change the wait time. But, friendly advice, unless you are in a hurry to get your account banned, you should not change this value.

**Comment by nanogirl21** [comment]
I'm in no hurry, but I think it would be nice to have the option to change that speed. Thank you.

**Comment by Kassadd** [comment]
After this commit, my exp drop from 40k/h to 3-4k/h. Seems each time I capture a pokemon, the PokemonOptimizer run and take 2 minutes or more...:

image

**Comment by nanogirl21** [comment]
@Anakin5 Can I use stardust to get full CP after evolving? can you add this maybe? Also seeking for way to find a way to keep 2 pokemon with different attacks and both with highest IV. Changing to Top 2 do not work.

**Comment by zx900930** [comment]
top: 2 seems not work

```
Traceback (most recent call last):
  File "pokecli.py", line 580, in <module>
    main()
  File "pokecli.py", line 105, in main
    bot.tick()
  File "E:\Program Files\PokemonGo-Bot\pokemongo_bot\__init__.py", line 482, in tick
    if worker.work() == WorkerResult.RUNNING:
  File "E:\Program Files\PokemonGo-Bot\pokemongo_bot\cell_workers\pokemon_optimizer.py", line 49, in work
    transfer, evo_best, evo_crap = self.get_family_optimized(family_id, family)
  File "E:\Program Files\PokemonGo-Bot\pokemongo_bot\cell_workers\pokemon_optimizer.py", line 82, in get_family_optimized
    evolve_best += self.get_top_rank(family, criteria)
  File "E:\Program Files\PokemonGo-Bot\pokemongo_bot\cell_workers\pokemon_optimizer.py", line 135, in get_top_rank
    worst = sorted_family[criteria.get("top", 1) - 1]
IndexError: list index out of range
```

**Pull request merged**